runtime/near-vm-logic/src/tests/gas_counter.rs

```diff
@@ -227,22 +227,26 @@ fn function_call_no_weight_refund() {
 fn test_overflowing_burn_gas_with_promises_gas() {
     let gas_limit = 3 * 10u64.pow(14);
     let mut logic_builder = VMLogicBuilder::default().max_gas_burnt(gas_limit);
-    let mut logic = logic_builder.build_with_prepaid_gas(gas_limit);
+    let mut logic = logic_builder.build(get_context_with_prepaid_gas(gas_limit));
+
+    let account_id = logic.internal_mem_write(b"rick.test");
+    let num_100u128 = logic.internal_mem_write(&100u128.to_le_bytes());
+    let num_10u128 = logic.internal_mem_write(&10u128.to_le_bytes());
 
     let index = promise_batch_create(&mut logic, "rick.test").expect("should create a promise");
-    logic.promise_batch_action_transfer(index, 100u128.to_le_bytes().as_ptr() as _).unwrap();
-    let call_id = logic.promise_batch_then(index, 9, "rick.test".as_ptr() as _).unwrap();
+    logic.promise_batch_action_transfer(index, num_100u128.ptr).unwrap();
+    let call_id = logic.promise_batch_then(index, account_id.len, account_id.ptr).unwrap();
 
     let needed_gas_charge = u64::max_value() - logic.gas_counter().used_gas() - 1;
     let function_name_len =
         needed_gas_charge / logic.config().ext_costs.cost(ExtCosts::read_memory_byte);
     let result = logic.promise_batch_action_function_call(
         call_id,
         function_name_len,
-        "x".as_ptr() as _,
+        0,
```

Comment:
I think it would probably be neater to allocate this as `dummy` and use `dummy.ptr/len` here rather than magic numbers.

Reply:
I dunno.  Here the dummy value is a function name with calculated length.  I kinda feel writing dummy function name will only introduce noise.  Not really sure.

I’ve changed the arguments though.